sanic/server.py

```diff
@@ -154,6 +157,8 @@ def data_received(self, data):
             self.headers = []
             self.parser = HttpRequestParser(self)
 
+        self.state['requests_count'] = self.state['requests_count'] + 1
```

Comment:
hmm, it depends on the needs/design. Both are fine. i'd prefer on_message_complete or data_received when it mainly starts process the request.